design-toolkit/components/src/components/table/index.tsx

```diff
@@ -0,0 +1,560 @@
+  type TableProps,
+} from './types';
+
+const { headerCellButton, pinIcon, rowCell, rowKebab, headerKebab } =
+  TableStyles();
+
+const dataTableCell = <T,>(cell: Cell<T, unknown>, persistent: boolean) => (
+  <TableCell
+    key={cell.id}
+    persistent={persistent}
+    narrow={cell.column.id === 'numeral' || cell.column.id === 'kebab'}
+    numeral={cell.column.id === 'numeral'}
+    kebab={cell.column.id === 'kebab'}
+    style={{ width: cell.column.getSize() }}
+  >
+    {flexRender(cell.column.columnDef.cell, cell.getContext())}
+  </TableCell>
+);
+
+const TableDefaultProps = {
+  kebabPosition: 'right',
+  persistRowKebabMenu: true,
+  persistHeaderKebabMenu: true,
+  persistNumerals: false,
+  pageSize: 10,
+  enableSorting: true,
+} as const;
+
+export function Table<T extends { id: string | number }>({
+  columns: columnsProp,
+  data: dataProp,
+  showCheckbox,
+  kebabPosition = TableDefaultProps.kebabPosition,
+  persistRowKebabMenu = TableDefaultProps.persistRowKebabMenu,
+  persistHeaderKebabMenu = TableDefaultProps.persistHeaderKebabMenu,
+  persistNumerals = TableDefaultProps.persistNumerals,
+  pageSize = TableDefaultProps.pageSize,
+  enableSorting = TableDefaultProps.enableSorting,
+  ...props
+}: TableProps<T>) {
+  const {
+    items: data,
+    moveAfter,
+    moveBefore,
+  } = useListData({
+    initialItems: dataProp,
+  });
+
+  const dataIds = useMemo<(string | number)[]>(
+    () => data?.map((item: T) => item.id),
+    [data],
+  );
+
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [hoveredArrow, setHoveredArrow] = useState<boolean>(false);
+
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+
+  /**
+   * moveUpSelectedRows moves the selected rows up in the table.
+   * It finds the first selected row, determines its index,
+   * and moves it before the previous row if it exists.
+   */
+  const moveUpSelectedRows = useCallback(() => {
+    const rowSelectionKeys = Object.keys(rowSelection).filter(
+      (id) => rowSelection[id],
+    );
+    const selectedRows = data.filter((item: any) =>
+      rowSelectionKeys.includes(item.id.toString()),
+    );
+
+    const firstSelectedRowId = selectedRows[0]?.id;
+    if (firstSelectedRowId) {
+      const rowIndex = dataIds.indexOf(firstSelectedRowId);
+
+      const prevRowId = dataIds[rowIndex ? rowIndex - 1 : 0];
+
+      if (prevRowId) {
+        moveBefore?.(prevRowId, rowSelectionKeys);
+      }
+    }
+  }, [data, dataIds, rowSelection, moveBefore]);
+
+  /**
+   * moveDownSelectedRows moves the selected rows down in the table.
+   * It finds the last selected row, determines its index,
+   * and moves it after the next row if it exists.
+   */
+  const moveDownSelectedRows = useCallback(() => {
+    const rowSelectionKeys = Object.keys(rowSelection).filter(
+      (id) => rowSelection[id],
+    );
+    const selectedRows = data.filter((item: any) =>
+      rowSelectionKeys.includes(item.id.toString()),
```

Comment:
In theory, we shouldn't run into cases where the `id` can be different types. Being specific about the `any` will help flush that out/force it.